src/libraries/System.Linq/src/System/Linq/Max.cs

```diff
@@ -341,6 +331,33 @@ public static double Max(this IEnumerable<double> source)
             return value;
         }
 
+        private static double Max(ReadOnlySpan<double> span)
+        {
+            if (span.IsEmpty)
+            {
+                ThrowHelper.ThrowNoElementsException();
+            }
+
+            int i;
+            for (i = 0; i < span.Length && double.IsNaN(span[i]); i++);
+
+            if (i == span.Length)
+            {
+                return span[^1];
+            }
+
+            double value;
+            for (value = span[i]; (uint)i < (uint)span.Length; i++)
+            {
+                if (span[i] > value)
```

Comment:
seems the code is intentionally not doing max the "proper" way
https://github.com/danmoseley/runtime/blob/9df283248256dd880eb1ee5ecce533ad206a55d3/src/libraries/System.Private.CoreLib/src/System/Math.cs#L893

Reply:
its doing a "pre-pass" above to skip until it finds the first non-NaN

and then relying on the fact that any future comparisons will return "false" for `NaN` and not find a replacement

there should probably be a comment explaining that here. Notably, the "pre-pass" could be vectorized, but I expect that wouldn't be beneficial in "practice" due to the unlikely case of encountering "many NaN".

we could also vectorize min/max, but this is again a case where there is some complications because you end up trading predicted branching for executing more code for for half the elements. -- That is, right now we are doing `span.Length` iterations with 1 branch per iteration. If we were vectorizing it, we'd be doing `span.Length / 2` iterations (for `Vector128<T>`), but we'd be doing 2 comparisons and a blend operation per iteration which might be worse.

Reply:
ah - I was wondering about that. Yes I think that's worth a comment.